### PR TITLE
feat: expose app accessibility status api

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Resolves https://github.com/electron/electron/issues/16090.

Exposes an api through System Preferences allowing an app to determine whether or not it's been designated a [trusted accessibility client](https://developer.apple.com/library/archive/documentation/Accessibility/Conceptual/AccessibilityMacOSX/), where that means an app that is allowed to modify the way users interact with their computer.

This API takes a prompt argument; if `false` is passed then the result will be returned silently, and if `true` is passed then a dialog (like the one below) will be shown.

screen shot 2018-12-17 at 8 32 33 pm

/cc @miniak @4ndv

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] PR release notes describe the change in a way relevant to app-developers

#### Release Notes

Notes: Exposes an API to allow apps to determine their status as a trusted accessibility client.

### CODE REVIEW

**Comment by miniak** [comment]
@codebytere you had a small bug in the code, passing `@YES` both when `prompt` was `true` and `false`. Fixed that + eliminated duplicate code by using conditional operator for the value.

**Comment by codebytere** [comment]
ah ok, i wasn't sure if objc bool literal could be coerced from c++ bools

File: docs/api/system-preferences.md [hunk 1]
```diff
@@ -327,6 +327,12 @@ You can use the `setAppLevelAppearance` API to set this value.
 Sets the appearance setting for your application, this should override the
 system default and override the value of `getEffectiveAppearance`.
 
+### `systemPreferences.isTrustedAccessibilityClient(prompt)` _macOS_
+
+* `prompt` Boolean - whether or not the user will be informed via prompt if the current process is untrusted.
+
+Returns `Boolean` - `true` if the current process is a trusted accessibility client and `false` if it is not.
```
Comment (BinaryMuse) [inline docs/api/system-preferences.md]: Does this return before or after the user is prompted? E.g. if the user opts to accept the prompt and add the app, will this return `true`?

Comment (codebytere) [reply docs/api/system-preferences.md]: the prompt can't be accepted per se, it just links to the system prefpane; so no :)

**Comment by fttx** [comment]
Will it be included in electron v4?

**Pull request merged**